src/zenml/exceptions.py

```diff
@@ -19,17 +19,42 @@
     from zenml.steps import BaseStepConfig
 
 
-class InitializationException(Exception):
+class ZenMLBaseException(Exception):
+    """Base exception for all ZenML Exceptions."""
+
+    def __init__(
+        self,
+        message: Optional[str] = None,
+        url: str = "https://docs.zenml.io",
+        override_message: bool = False,
```

Comment:
Maybe we can rename this to `include_url` or something more to the point of what it actually does? I'm also not quite sure if it should be enabled or disabled by default, not sure how much of a help the docs are in specific errors like this. To me they cover a broader scope and most of these errors refer to specific details that won't be mentioned in the docs.